src/components/Input/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { TextField } from "@radix-ui/themes"
+import clsx from "clsx"
+import type { ReactNode } from "react"
+import type { FieldValues, Path, UseFormRegister } from "react-hook-form"
+import styles from "./styles.module.css"
+
+type Props<T, TFieldValues extends FieldValues> = {
+  name: string
+  register?: UseFormRegister<TFieldValues>
+  placeholder?: string
+  value: string
+  fullWidth?: boolean
+  disabled?: boolean
+  onChange?: (value: string) => void
+  className?: string
+  slotLeft?: ReactNode | null
+  slotRight?: ReactNode | null
+  type?: "text" | "number"
+}
+export const Input = <T extends string, TFieldValues extends FieldValues>({
+  name,
+  register,
+  value,
+  placeholder,
+  fullWidth = false,
+  disabled = false,
+  onChange,
+  className,
+  slotLeft = null,
+  slotRight = null,
+  type = "text",
+}: Props<T, TFieldValues>) => {
+  const registerProps = register ? register(name as Path<TFieldValues>) : {}
+  const { onChange: formOnChange, ...rest } = registerProps as {
+    onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void
+  }
```

Comment:
We have the same issue here, we should not bind our inputs to `react-hook-form`. I actually want to get rid of react-hook-form from swap widget, it doesn't bring anything to it.